processor/configuration/src/test/java/com/redhat/service/smartevents/processor/GatewayConfiguratorImplTest.java

```diff
@@ -0,0 +1,175 @@
+import com.redhat.service.smartevents.processor.actions.webhook.WebhookAction;
+import com.redhat.service.smartevents.processor.actions.webhook.WebhookActionValidator;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@QuarkusTest
+class GatewayConfiguratorImplTest {
+
+    private static final Map<String, ExpectedBeanClasses<Action>> EXPECTED_ACTION_BEANS = Map.of(
+            KafkaTopicAction.TYPE, expect(KafkaTopicActionValidator.class, null, null),
+            SendToBridgeAction.TYPE, expect(SendToBridgeActionValidator.class, SendToBridgeActionResolver.class, null),
+            SlackAction.TYPE, expect(SlackActionValidator.class, SlackActionResolver.class, SlackActionConnector.class),
+            WebhookAction.TYPE, expect(WebhookActionValidator.class, null, null));
+
+    private static final Map<String, ExpectedBeanClasses<Source>> EXPECTED_SOURCE_BEANS = Collections.emptyMap();
+
+    @Inject
+    GatewayConfiguratorImpl configurator;
+
+    @Test
+    void testExpectedActionBeans() {
+        for (Map.Entry<String, ExpectedBeanClasses<Action>> entry : EXPECTED_ACTION_BEANS.entrySet()) {
+            String type = entry.getKey();
+            ExpectedBeanClasses<Action> expected = entry.getValue();
+
+            assertThat(configurator.getActionValidator(type))
+                    .as("GatewayConfigurator.getActionValidator(\"%s\") should not return null", type)
+                    .isNotNull();
+            assertThat(configurator.getActionValidator(type))
+                    .as("GatewayConfigurator.getActionValidator(\"%s\") should be instance of %s", type, expected.validatorClass.getSimpleName())
+                    .isInstanceOf(expected.validatorClass);
+
+            assertThat(configurator.getActionResolver(type))
+                    .as("GatewayConfigurator.getActionResolver(\"%s\") should not return null", type)
+                    .isNotNull();
+
+            if (expected.resolverClass == null) {
+                assertThat(configurator.getActionResolver(type))
+                        .as("GatewayConfigurator.getActionResolver(\"%s\") should be empty", type)
+                        .isNotPresent();
+            } else {
+                assertThat(configurator.getActionResolver(type))
+                        .as("GatewayConfigurator.getActionResolver(\"%s\") should not be empty", type)
+                        .isPresent();
+                assertThat(configurator.getActionResolver(type))
+                        .as("GatewayConfigurator.getActionResolver(\"%s\") should contain instance of %s", type, expected.resolverClass.getSimpleName())
+                        .containsInstanceOf(expected.resolverClass);
+            }
+
+            assertThat(configurator.getActionConnector(type))
+                    .as("GatewayConfigurator.getActionConnector(\"%s\") should not return null", type)
+                    .isNotNull();
```

Comment:
similar question as above

Reply:
Same [answer](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/646#discussion_r855169944) (just replace _Resolver_ with _Connector_ but the meaning is the same)